server/api/controller/plan_status_change.js

```diff
@@ -0,0 +1,99 @@
+const Controller = require('./controller');
+const PlanStatusChange = require('../model/plan_status_change');
+const Plan = require('../model/plan');
+const Log = require('../lib/log');
+const { debug } = require('../lib/log');
+
+class PlanStatusChangeController extends Controller {
+
+	/**
+	 * Return status for this plan.
+	 * @param {IncomingRequest} req
+	 */
+	async byPlan (req) {
+		const planId =req.params.id;
+		const PLAN_TYPE_77_78 = "הודעה לפי סעיף 77 ו 78 לחוק התכנון והבניה";
+		let planType, planCreatedAt;
+		planType=null;
+
+		await Plan.fetchByPlanID(planId)
+		.then(plan => {
+			if (plan) {
+				planType = (plan.attributes.data.ENTITY_SUBTYPE_DESC);
+				planCreatedAt = plan.attributes.created_at;
+				Log.debug('planType ', planType, 'planCreatedAt ', planCreatedAt);
+		}})
+		.catch(error => {
+			Log.error('Plan.fetchByPlanID failed to fetch plan details');
+			Log.error(error.message);
+			Log.error(error.stack);
+			
+		});
+
+		if (!planType) {
+			return {"cancellationDate": null, "steps": []};;
+		}
+
+		return this.model.byPlan(planId).then(collection => {
+			Log.debug(this.tableName, 'Get status list', req.params.id);
+			
+			let steps = [];
+			/* If the plan type (סוג תוכנית) = הודעה לפי סעיף 77 ו 78 לחוק התכנון והבניה, add a stepId:0, 
+			set to date: plan.createddate, completed:true, current:true, and all other steps to false for both. */
+			if (planType===PLAN_TYPE_77_78) {
+				steps = [{name:'פרסום על הכנת תוכנית',description:'הודעה לפי סעיף 77 ו 78 לחוק התכנון והבניה',stepId: 0, completed: true, current: true, date: planCreatedAt}];
+				Log.debug(`Plan type is '${PLAN_TYPE_77_78}' so setting step zero as `, steps);
+			}
+
+			// if date is not null, set completed to true
+			steps = steps.concat(collection.map(function(element){
```

Comment:
Please export this logic to be a static/ normal function on the Plan Model, therefore leaving the controller logic free.

Reply:
I.e :
```
// Plan.js
Plan.getStatusChanges = (planId) => {//logic ... }
```